selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/stacks/Stacks.java

```diff
@@ -0,0 +1,231 @@
+    String STACKS_LIST_HEADER_XPATH = "//md-item[@class='noselect']//span";
+    String SORT_STACKS_BY_NAME_BUTTON_XPATH = "//span[contains(@class, 'header-sort-direction')]";
+    String BULK_CHECKBOX_XPATH = "//md-checkbox[@aria-label='Stack list']";
+    String NO_STACKS_FOUND_XPATH = "//span[text()='No stacks found.']";
+    String STACK_ITEM_XPATH = "//div[contains(@class, 'stack-item-name')]";
+    String STACK_ITEM_CHECKBOX_XPATH_PATTERN = "//div[@id='stack-name-%s']//md-checkbox";
+    String STACK_ITEM_NAME_XPATH_PATTERN = "//div[@id='stack-name-%s']";
+    String STACK_ITEM_NAME_CONTAINS_XPATH_PATTERN = "//div[contains(@id,'stack-name-%s')]";
+    String STACK_ITEM_COMPONENTS_XPATH_PATTERN =
+        "//div[@id='stack-name-%s']//div[contains(@class,'stack-item-components')]";
+    String STACK_ITEM_DESCRIPTION_XPATH_PATTERN =
+        "//div[@id='stack-name-%s']//div[contains(@class,'stack-item-description')]";
+    String STACK_ITEM_DELETE_BUTTON_XPATH_PATTERN =
+        "//div[@id='stack-name-%s']//a[@uib-tooltip='Delete Stack']";
+    String DUPLICATE_STACK_BUTTON_XPATH_PATTERN =
+        "//div[@id='stack-name-%s']//a[@uib-tooltip='Duplicate stack']";
+    String BUILD_STACK_FROM_RECIPE_DIALOG_XPATH = "//*[@title='Build stack from recipe']";
+  }
+
+  @FindBy(id = Locators.TOOLBAR_ID)
+  WebElement toolbar;
+
+  @FindBy(id = Locators.ADD_STACK_BUTTON_ID)
+  WebElement addWorkspaceBtn;
+
+  @FindBy(id = Locators.DELETE_STACK_BUTTON_ID)
+  WebElement deleteStackButton;
+
+  @FindBy(xpath = Locators.BULK_CHECKBOX_XPATH)
+  WebElement bulkCheckbox;
+
+  @FindBy(xpath = Locators.BUILD_STACK_FROM_RECIPE_BUTTON_XPATH)
+  WebElement buildStackFromRecipeButton;
+
+  @FindBy(id = Locators.DELETE_DIALOG_BUTTON_ID)
+  WebElement deleteDialogBtn;
+
+  @FindBy(xpath = Locators.SEARCH_STACK_FIELD_XPATH)
+  WebElement searchStackField;
+
+  public void waitToolbarTitleName() {
+    seleniumWebDriverHelper.waitVisibility(toolbar);
+  }
+
+  public void waitAddStackButton() {
+    seleniumWebDriverHelper.waitVisibility(addWorkspaceBtn);
+  }
+
+  public void clickOnAddStackButton() {
+    seleniumWebDriverHelper.waitAndClick(addWorkspaceBtn);
+  }
+
+  public void waitBuildStackFromRecipeButton() {
+    seleniumWebDriverHelper.waitVisibility(buildStackFromRecipeButton);
+  }
+
+  public void clickOnBuildStackFromRecipeButton() {
+    seleniumWebDriverHelper.waitAndClick(buildStackFromRecipeButton);
+  }
+
+  public void waitBuildStackFromRecipeDialog() {
+    seleniumWebDriverHelper.waitVisibility(By.xpath(Locators.BUILD_STACK_FROM_RECIPE_DIALOG_XPATH));
+  }
+
+  public void selectAllStacksByBulk() {
+    seleniumWebDriverHelper.waitAndClick(bulkCheckbox);
+  }
+
+  public void selectStackByCheckbox(String stackName) {
+    seleniumWebDriverHelper.waitAndClick(
+        By.xpath(String.format(Locators.STACK_ITEM_CHECKBOX_XPATH_PATTERN, stackName)));
+  }
+
+  public boolean isStackChecked(String workspaceName) {
+    String attrValue =
```

Comment:
I would propose to move the code to separate method of **SeleniumWebDriverHelper** class similar to **waitVisibilityAndGetValue**.